Target `spicy-driver` defined multiple times

We currently define the executable target `spicy-driver` twice in `CMakeLists.txt`. This seems to be an issue e.g., when building with `HILTI_USE_JIT` defined.
```
CMake Error at spicy/CMakeLists.txt:192 (add_executable):
  add_executable cannot create target "spicy-driver" because another target
  with the same name already exists.  The existing target is an executable
  created in source directory "/build/spicy/spicy".  See documentation for
  policy CMP0002 for more details.
```

We should probably just remove the conditional target under `HILTI_USE_JIT`.

--- a/spicy/CMakeLists.txt
+++ b/spicy/CMakeLists.txt
@@ -188,12 +188,6 @@ add_custom_command(TARGET spicy-build
                    POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/bin/spicy-build
                                                                ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
 
-if ( HILTI_USE_JIT )
-    add_executable(spicy-driver bin/spicy-driver.cc)
-    target_link_libraries(spicy-driver PRIVATE spicy)
-endif ()
-
-
 ## Tests
 add_executable(spicy-tests
                tests/main.cc

